src/components/Dependencies/Dependencies.tsx

```diff
@@ -1,117 +1,78 @@
 
-  public render(): ReactElement<HTMLElement> {
-    const { name, version } = this.state;
-    const tagText = `${name}@${version}`;
-    return <Tag className={'dep-tag'} clickable={true} label={tagText} onClick={this.handleOnClick} />;
-  }
+  function handleClick(name: string): void {
+    enableLoading && enableLoading();
 
-  private handleOnClick = () => {
-    const { name } = this.state;
-    const { onLoading, history } = this.props;
-
-    onLoading && onLoading();
     history.push(`/-/web/detail/${name}`);
-  };
-}
+  }
 
-const WrapperDependencyDetail = withRouter(DepDetail);
+  return (
+    <CardWrap>
+      <CardContent>
+        <Heading variant="subtitle1">{`${title} (${deps.length})`}</Heading>
+        <Tags>
+          {deps.map(([name, version]) => (
+            // eslint-disable-next-line
+            <Tag className={'dep-tag'} clickable={true} key={name} label={`${name}@${version}`} onClick={() => handleClick(name)} />
+          ))}
+        </Tags>
+      </CardContent>
+    </CardWrap>
+  );
+};
+
+function hasKeys(object?: { [key: string]: any }): boolean {
+  return !!object && Object.keys(object).length > 0;
+}
 
-class DependencyBlock extends Component<{ title: string; dependencies: [] }> {
-  public render(): ReactElement<HTMLElement> {
-    const { dependencies, title } = this.props;
-    const deps = Object.entries(dependencies) as [];
+const Dependencies: React.FC<{}> = () => {
+  const { packageMeta } = useContext(DetailContext);
 
-    return (
-      <DetailContextConsumer>
-        {({ enableLoading }) => {
-          return (
-            <CardWrap>
-              <CardContent>
-                <Heading variant="subtitle1">{`${title} (${deps.length})`}</Heading>
-                <Tags>{this.renderTags(deps, enableLoading)}</Tags>
-              </CardContent>
-            </CardWrap>
-          );
-        }}
-      </DetailContextConsumer>
-    );
+  if (!packageMeta) {
+    throw new Error('DetailContext is required');
   }
 
-  private renderTags = (deps: [], enableLoading?: () => void) =>
-    deps.map(dep => {
-      const [name, version] = dep as [string, string];
+  const { latest } = packageMeta;
+  // FIXME: add dependencies to package meta type
+  // @ts-ignore
+  const { dependencies, devDependencies, peerDependencies, name } = latest;
+  const dependencyMap = { dependencies, devDependencies, peerDependencies };
+  const hasDependencies = hasKeys(dependencies) || hasKeys(devDependencies) || hasKeys(peerDependencies);
 
-      return <WrapperDependencyDetail key={name} name={name} onLoading={enableLoading} version={version} />;
-    });
-}
-
-class Dependencies extends Component {
-  public render(): ReactElement<HTMLElement> {
+  if (hasDependencies) {
     return (
-      <DetailContextConsumer>
-        {packageMeta => {
-          return this.renderDependencies(packageMeta as VersionPageConsumerProps);
-        }}
-      </DetailContextConsumer>
+      <Fragment>
```

Comment:
```suggestion
      <>
```

Reply:
Why are we using `<Fragment>` instead of `<>` ? `<>` is a shortcut from `Fragment` but it is the same :)

Reply:
It's just a suggestion :-) I prefer `<>`

Reply:
Oh yes sure I thought I wrote the other way ! I'll fix it tomorrow

Reply:
haha I prefer `<Fragment>`, :P but honestly I started to like `<>`, so +1 for @priscilawebdev